packages/components/psammead-content-shift-blocker/src/index.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import when from 'ramda/src/when';
+import is from 'ramda/src/is';
+import { number, string, node, oneOfType } from 'prop-types';
+import styled from 'styled-components';
+
+const convertToPixels = num => `${num}px`;
+
+const getSize = when(is(Number), convertToPixels);
+
+const getHeight = ({ wrapperHeight }) => getSize(wrapperHeight);
+
+const getWidth = ({ wrapperWidth }) => getSize(wrapperWidth);
+
+const Wrapper = styled.div`
+  overflow: auto;
+  height: ${getHeight};
+  width: ${getWidth};
+`;
+
+const initIntersectionObserver = ({ wrapperEl, setWrapperIO }) => {
+  const init = () => {
+    const callback = ([wrapperEntry]) => {
+      setWrapperIO(wrapperEntry);
+    };
+    const IO = new IntersectionObserver(callback);
+
+    IO.observe(wrapperEl.current);
+
+    return IO;
+  };
+
+  if ('IntersectionObserver' in window) {
+    return init();
+  }
+
+  return import(
+    /* webpackChunkName: "intersection-observer-polyfill" */ 'intersection-observer'
+  ).then(() => {
+    IntersectionObserver.prototype.POLL_INTERVAL = 100;
+    return init();
+  });
+};
+
+const initResizeObserver = ({ wrapperEl, setContentElRect }) => {
+  const init = ResizeObserver => {
+    const callback = ([contentEntry]) => {
+      setContentElRect(contentEntry.contentRect);
+    };
+    const RO = new ResizeObserver(callback);
+
+    RO.observe(wrapperEl.current.firstChild);
+
+    return RO;
+  };
+
+  if ('ResizeObserver' in window) {
+    return init(ResizeObserver);
+  }
+  return import(
+    /* webpackChunkName: "resize-observer-polyfill" */ '@juggle/resize-observer'
+  ).then(module => {
+    const ResizeObserver = module.default;
+    return init(ResizeObserver);
+  });
+};
+
+const isScrollAnchoringSupported = () => {
+  // https://drafts.csswg.org/css-scroll-anchoring/
+  if ('CSS' in window) {
+    return CSS.supports('overflow-anchor', 'auto');
+  }
+  return false;
+};
+
+const ContentShiftBlocker = ({ children, initialHeight, initialWidth }) => {
+  const wrapperEl = useRef(null);
+  const scrollHeight = useRef(null);
+  const scrollAnchoringIsSupported = useRef(null);
+  const [wrapperIO, setWrapperIO] = useState({});
+  const [contentElRect, setContentElRect] = useState({});
+  const [wrapperDimensions, setWrapperDimensions] = useState({
+    height: initialHeight,
+    width: initialWidth,
+  });
+
+  useEffect(() => {
+    // component did mount
```

Comment:
to be accurate I think it would be 
```suggestion
    // component did mount/update
```
Unsure if the comments are needed?

Reply:
@andrewscfc because the dependencies are specified and they are empty (empty array as 2nd argument) then this useEffect will only ever be called after the first render. If the empty array was not passed in then it would be component did mount/update event. See https://dev.to/trentyang/replace-lifecycle-with-hooks-in-react-3d4n or a better explanation